pkg/controllers/gacha-controller.go

```diff
@@ -99,7 +87,12 @@ func UpdateUser(w http.ResponseWriter, r *http.Request) {
 }
 
 func ListCharacters(w http.ResponseWriter, r *http.Request) {
-
+	user, ok := r.Context().Value("user").(models.User)
+	if !ok {
+		http.Error(w, "User not found", http.StatusUnauthorized)
```

Comment:
What would be the situation without users?

Reply:
@GuiltyMorishita I think we would always require a user to be authenticated before listing the characters. So if there are no users then nobody can see the characters present.

Reply:
@iamananya 
Assuming that the existence of User is guaranteed by the authentication middleware, shouldn't the error that should be returned if User is empty be 5xx instead of 4xx? This is what I wanted to show.

Reply:
Okay @GuiltyMorishita , I understood .SO I need to update the status code. I will resolve this issue.

Reply:
Resolved in PR #13